core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -6357,6 +6357,144 @@ LogicalProject(EMPNO=[$3], ENAME=[$4], JOB=[$5], MGR=[$6], HIREDATE=[$7], SAL=[$
 LogicalJoin(condition=[true], joinType=[inner])
   LogicalTableScan(table=[[scott, EMP]])
   LogicalTableScan(table=[[scott, DEPT]])
+]]>
```

Comment:
these tests are very hard for humans to certify.
it would be very nice if we could execute these plans somehow on some input data to compare results with the non-expanded form.

Reply:
This is a very good suggestion. I also thought about it when writing the test. Is there any existing test framework that can support this kind of logic? I don’t seem to have found it. Do I need to make an extension for JdbcTest?

Reply:
Maybe CalciteASsert.consistentResult is the function needed.

Reply:
Thanks, I'll see how to adjust the test.

Reply:
I found a way to verify it, do you think it works?